Zeeguu Reader/Zeeguu Reader/ZGTextView.swift

```diff
@@ -48,13 +51,31 @@ class ZGTextView: UITextView {
 	
 	override func canPerformAction(action: Selector, withSender sender: AnyObject?) -> Bool {
 		if action == "translate:" {
-			translate(self)
+			if (willInstantlyTranslate) {
+				translate(self)
+				return false
+			}
+			return true
 		}
 		return false
 	}
 	
 	func translate(sender: AnyObject?) {
+		if (isTranslating) {
+			return
+		}
+		isTranslating = true
 		print("translate called for \(self.selectedText()) with context: \"\(self.selectedTextContext())\"")
+		
+		let range = self.selectedRange
+//		let firstCharAfterRange = NSMakeRange(range.location + range.length, 1)
+		
+		let attributes = self.attributedText.attributesAtIndex(range.location + range.length, effectiveRange: nil)
+		print("attributes: \(attributes)")
+		if let color = attributes[NSForegroundColorAttributeName] where color.isEqual(UIColor.lightGrayColor()) {
+			self.selectedTextRange = nil
+			return
+		}
 		ZeeguuAPI.sharedAPI().translateWord(self.selectedText(), title: article.title, context: self.selectedTextContext(), url: article.url) { (dict) -> Void in
 			if let t = dict?["translation"].string {
```

Comment:
`dict` is a `JSON` object. See the Zeeguu API for the `SiftyJSON.swift` file that contains its definition. Also look at https://github.com/SwiftyJSON/SwiftyJSON